docs/pipelines/overview/configuration.md

```diff
@@ -1,95 +1,368 @@
-# Pipelines Configuration
+# Pipelines Configurations as Code
 
+:::note
+This page includes documentation for HCL Pipelines Configuration as Code.
 
-:::warning This page is under active development and is a work in progress, June 2024
+If you are using an older version of Pipelines, please refer to the documentation on [Upgrading from config.yml](../upgrading/upgrading-from-config-yml.md).
 :::
 
-Pipelines behavior is primarily configured via a yaml file at `.gruntwork/config.yml` in your repository. Pipelines also reads [.mise.toml](https://docs.gruntwork.io/pipelines/upgrading/upgrading-from-infrastructure-pipelines#adding-misetoml) at the root of the repository to determine what version of OpenTofu/Terraform and Terragrunt to use.
+:::caution
+As of July, 2024, the YAML configuration is still the primary mechanism for configuring Pipelines in DevOps Foundations. The best experience in DevOps Foundations remains with the YAML configuration system, as interactions with Account Factory still need improvement in the HCL configuration system.
```

Comment:
As I'm reading this, maybe it makes more sense to more away from the notion that HCL is primary, not even reference migration, and instead brand the HCL configs "experimental" or "beta" or some such, and frame it the opposite way - this HCL config is specifically for brownfield support, and is not intended for all customers yet.

Reply:
Moved around stuff to make it work more like that.